.github/workflows/build-cppfront.yaml

```diff
@@ -0,0 +1,56 @@
+name: Multi-platform Build of cppfront
+on: push
+jobs:
+  build-windows:
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ilammy/msvc-dev-cmd@v1
+      - name: Compiler name & version
+        run: cl.exe
+      - name: Build
+        run: cl.exe source/cppfront.cpp -std:c++latest -MD -EHsc -experimental:module -W4
+  build-unix-like:
+    strategy:
+      fail-fast: false
+      matrix:
+        runs-on: [ubuntu-latest, macos-latest]
+        compiler: [g++-10, g++-11, g++-12, clang++-12, clang++-14, clang++]
+        cxx-std: ['c++20', 'c++23']
+        exclude:
+          # Unversioned clang++ doesn't run on ubuntu
+          - runs-on: ubuntu-latest
+            compiler: clang++
```

Comment:
Besides this one,
I think you can remove all other exclusions
by changing `c++23` to `c++2b`,
removing `macos-latest` from `runs-on`,
and adding
```yaml
        include:
          - runs-on: macos-latest
            compiler: clang++
```

Reply:
I did some testing by changing `c++23` to `c++2b` and managed to simplify the matrix by using `include` as suggested. Summary:

- `gcc-10` doesn't recognize either `c++2b` or `c++23`
- `clang++-12` does recognize `c++2b` but fails to compile, doesn't recognize `c++23`
- `clang++-14` same as above

The only exclusion now is `gcc-10` on the `c++2b` configuration, also, I have attached the build logs of the failed builds since this doesn't seem intentional (why would it compile with `c++20` but not `c++2b`?).

[7_build-unix-like (ubuntu-latest, clang++-12, c++2b).txt](https://github.com/hsutter/cppfront/files/11534864/7_build-unix-like.ubuntu-latest.clang%2B%2B-12.c%2B%2B2b.txt)
[9_build-unix-like (ubuntu-latest, clang++-14, c++2b).txt](https://github.com/hsutter/cppfront/files/11534884/9_build-unix-like.ubuntu-latest.clang%2B%2B-14.c%2B%2B2b.txt)

Reply:
I am not sure about using `c++2b` with compiler versions that do not support `c++23` (as I would be with `c++2a` instead of `c++20`). There is a good reason for this lack of support - the standard is not fully implemented in the particular compiler version. A build failure might actually be the result of shortcomings of the compiler rather than issues in the code. It is true there might be bugs even with full support, but those are much less likely.

Reply:
I will leave up to Herb to decide if we support partial implementations, what I find weird here is that `c++2b` > `c++20`, yet `c++20` compiles but `c++2b` does not. I would expect for cppfront to still compile if the user opts-in to a partial implementations that is newer than `c++20`...

Reply:
The issue seems to be <https://github.com/llvm/llvm-project/issues/58206>.

Reply:
As I said previously, I was pretty sure this is a compiler bug not an issue in the code.

Reply:
Do we disable those 2 explicitly then for the time being? I can reference the issue in a comment in that case.

Reply:
Please, do.

Reply:
Done!

Reply:
I was able to compile Cppfront with <https://github.com/llvm/llvm-project/commit/37b474a20e8a731bec9db7033e02013c07bd6fa4>.

Reply:
Opened #818 to workaround <https://github.com/llvm/llvm-project/pull/65456>.